From d51d36e40fe5b15a16c49267d14b25551f32427b Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Tue, 2 Oct 2018 16:49:28 +0300
Subject: [PATCH 220/741] sdk_dpaa: ceetm: lower the default congestion
 thresholds

commit d8b9e50e83497eca0b46a94ad011f64c0c0ca64f from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

The congestion thresholds need to be set in such a way that:
    a) the threshold is high enough so that frames aren't dropped
    unnecessarily
    b) the threshold is low enough so that the latency isn't too big

The current thresholds are set too high. In forwarding scenarios, the
latency is too large and frames are dropped on ingress due to a lack of
buffers.

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/Kconfig b/drivers/net/ethernet/freescale/sdk_dpaa/Kconfig
index b37b2a001205..beed824b67d1 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/Kconfig
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/Kconfig
@@ -24,7 +24,7 @@ config FSL_DPAA_CEETM_CCS_THRESHOLD_1G
 	hex "CEETM egress congestion threshold on 1G ports"
 	depends on FSL_DPAA_CEETM
 	range 0x1000 0x10000000
-	default "0x000a0000"
+	default "0x00005000"
 	help
 	  The size in bytes of the CEETM egress Class Congestion State threshold on 1G ports.
 	  The threshold needs to be configured keeping in mind the following factors:
@@ -38,7 +38,7 @@ config FSL_DPAA_CEETM_CCS_THRESHOLD_10G
 	hex "CEETM egress congestion threshold on 10G ports"
 	depends on FSL_DPAA_CEETM
 	range 0x1000 0x20000000
-	default "0x00640000"
+	default "0x00032000"
 	help
 	  The size in bytes of the CEETM egress Class Congestion State threshold on 10G ports.
 	  See FSL_DPAA_CEETM_CCS_THRESHOLD_1G for details.
-- 
2.17.1

